contracts/tokenLocking/TokenLocking.sol

```diff
@@ -172,19 +172,14 @@ contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
     // Transfer the the tokens
     Lock storage userLock = userLocks[_token][_user];
     userLock.balance = sub(userLock.balance, _amount);
-    makeConditionalDeposit(_token, _amount, _recipient);
+
+    require(ERC20Extended(_token).transfer(_recipient, _amount), "colony-token-locking-transfer-failed");
```

Comment:
The change is to avoid indeterminacy in the destination of the tokens. If we had `makeConditionalDeposit`, then depending on the state of the external world (the locks), the claimed stake would either remain in the user's tokenLocking account, or be held by the address directly. This means that if a contract flow involves sending those tokens elsewhere, the contract would need to check to see where the tokens went (involving some `balanceBefore`, `balanceAfter` ugliness). Previously, the tokens would always be _somewhere_ in TokenLocking, meaning a call to `claim` could sort it out in either case. But there's no one-click reconciliation available now, so I thought it best to provide determinacy by always sending tokens to the obligator.

Reply:
We want it to be easy for users to leave their tokens in this contract though. They shouldn't be forced to redeposit unless they have to. That was the whole point of the original design.

You're right that there's indeterminacy in the destination of the tokens, but I don't think that removing it by always having the tokens leave the contract is the correct solution - I think any contracts interacting with this should be able to cope. Is `deposit(token, 0, true)` prior to making the corresponding call not equivalent to `claim(token, true)` after the corresponding call?

Reply:
Hey! That's a fine idea.